### ISSUE TITLE
Support specifying Secret key (filename) on IAMServiceAccountKey .

### ISSUE THREAD

**Issue by ismailbaskin** [GoogleCloudPlatform/k8s-config-connector#42]
IAMServiceAccountKey creates a Secret with `key.json` key. Many helm packages needs spesific key for google service account key such as `google-credentials.json` or `credentials.json`. I think the option of specifying the file name will be quite useful.

* Example : [external-dns ](https://github.com/helm/charts/blob/master/stable/external-dns/templates/deployment.yaml#L223) `credentials.json` filename .

**Comment by AlexBulankou** [comment]
Thanks for suggestion, @ismailbaskin! Today `key.json` is hard-coded. Marking as enhancement.

**Comment by AlexBulankou** [comment]
@ismailbaskin, further follow up: looking into this issue it would make sense for [external-dns helm chart that you're referencing](https://github.com/helm/charts/blob/master/stable/external-dns/templates/deployment.yaml#L223), to support specifying the file name through the values so that users are not forced to use a specific name of their credentials file.
Another approach would be to have a step in your workload that gets the secret that was created as yaml and then changes `key.json` to a different name and reapplies.
Would any of these work for you?

**Comment by ismailbaskin** [comment]
Are you suggesting the second approach as a workaround? Our aim is having a git repository with full Kubernetes yaml files without any hack scripts for concrete gitops experience.
 
To be honest I'd rather be able to specify the key file name. It would be much more developer-friendly.

**Comment by AlexBulankou** [comment]
@ismailbaskin , agreed that the second approach (custom workflow to change key.json to a different name) is adding complexity. What do you think about the first approach, which would be preferred, I think - support specifying file name through the values in the Helm chart?

**Comment by ismailbaskin** [comment]
@AlexBulankou The first approach perfectly fits with my problem.

**Comment by jo2y** [comment]
Changing every other piece of software that doesn't use key.json as the filename doesn't seem very scalable. My use-case is reading from a private image repository.  The file name expected there is .dockerconfigjson.

https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials

**Comment by AlexBulankou** [comment]
@jo2y , thanks for pointing out the use case for reading image from a private image repository. 
Let's clarify this use case. I don't think it will be sufficient to simply allow changing  the file name from `key.json` to `.dockerconfigjson` and still use the same service account key secret format that Config Connector generates, the key needs to be of type `kubernetes.io/dockerconfigjson`. Is this correct or you found that simply changing `key.json` to `.dockerconfigjson` is sufficient to have it working?

Today, a validated way to create a secret to pull image based on Config Connector's IAMServiceAccountKey is have this additional step to generate docker-registry secret from the secret generated by IAMServiceAccountKey.

```bash
kubectl create secret docker-registry gcr-docker-key \
  --docker-server=https://gcr.io \  # example for gcr.io
  --docker-username=_json_key \
  --docker-email=user@example.com \ # any valid email format
  --docker-password="$(kubectl get secret gcr-sa-key -o go-template=$'{{index .data "key.json"}}' | base64 --decode)" # gcr-sa-key is Config Connector IAMServiceAccountKey name.
```

**Comment by ismailbaskin** [comment]
@jo2y I don't know who the author is 😉, but here's a nice article about it. https://www.keeprunning.io/2019/10/18/gcr-image-from-external-kubernetes/

**Comment by matheuscscp** [comment]
Just filed a PR for this, all we need is a new annotation: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/4595
